packages/next/src/build/webpack/plugins/next-types-plugin.ts

```diff
@@ -673,24 +675,26 @@ export class NextTypesPlugin {
 
           await Promise.all(promises)
 
+          // Support `"moduleResolution": "Node16" | "NodeNext"` with `"type": "module"`
+
+          const packageJsonAssetPath = path.join(
+            assetDirRelative,
+            'types/package.json'
+          )
+
+          assets[packageJsonAssetPath] = new sources.RawSource(
+            '{"type": "module"}'
+          ) as unknown as webpack.sources.RawSource
+
           if (this.typedRoutes) {
             if (this.dev && !this.isEdgeServer) {
               devPageFiles.forEach((file) => {
                 this.collectPage(file)
               })
             }
 
-            // Support tsconfig values for "moduleResolution": "Node16" or "NodeNext"
-            const packageJsonTypePath = path.join('types', 'package.json')
-            const packageJsonAssetPath =
-              assetDirRelative + '/' + normalizePathSep(packageJsonTypePath)
-            assets[packageJsonAssetPath] = new sources.RawSource(
-              '{"type": "module"}'
-            ) as unknown as webpack.sources.RawSource
+            const linkAssetPath = path.join(assetDirRelative, 'types/link.d.ts')
```

Comment:
Same here

Reply:
I thought both work? I've always used this even when I was using Windows...

Reply:
Ok! I want to just make sure

Reply:
@shuding Not related, but can I also get a review for #48723 as well? Thanks!